cpp/density/GaussianDensity.cc

```diff
@@ -31,94 +31,107 @@ vec3<unsigned int> GaussianDensity::getWidth()
     return m_width;
 }
 
-//! internal
-/*! \brief Function to compute the density array
- */
+//! Compute the density array.
 void GaussianDensity::compute(const freud::locality::NeighborQuery* nq)
 {
-    auto box = nq->getBox();
+    // Don't allow 3D boxes after computing in 2D (m_width is altered)
+    if (!nq->getBox().is2D() && m_box.is2D())
```

Comment:
Should we also forbid the other direction, i.e. 
```suggestion
    if (nq->getBox().is2D() != m_box.is2D())
```
?

Reply:
We should have each instance of `SphereVoxelization` and `GaussianDensity` only perform the calculation in a single number of dimensions. It removes the ambiguity of what is and isn't allowed in terms of going between numbers of dimensions between calculations and prevents any unseen errors associated with that that we might be missing.